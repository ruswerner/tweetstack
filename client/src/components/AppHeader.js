import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import AppBar from 'material-ui/AppBar';
import FlatButton from 'material-ui/FlatButton';
import Logo from './Logo';

export default class AppHeader extends PureComponent {

  static propTypes = {
    onClearAll: PropTypes.func.isRequired
  };

  render() {
    return (
      <AppBar title="TWEETSTACK"
              iconElementLeft={<Logo/>}
              iconElementRight={<FlatButton id="clear-all" label="Clear All" onClick={this.props.onClearAll}/>}
              style={{flexShrink: 0}}
      />
    );
  }

}
