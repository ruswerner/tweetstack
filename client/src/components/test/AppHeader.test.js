import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppHeader from '../AppHeader';

const noop = () => null;

describe('AppHeader', function() {

  describe('when button is clicked', function() {
    it('should call onClearAll', function() {
      const onClearAll = sinon.spy();
      const wrapper = mount(
        <MuiThemeProvider>
          <AppHeader onClearAll={onClearAll}/>
        </MuiThemeProvider>
      );
      wrapper.find('button[id="clear-all"]').simulate('click');
      expect(onClearAll.callCount).toEqual(1);
    });
  });

  it('should match snapshot', function() {
    const tree = renderer.create(
      <MuiThemeProvider>
        <AppHeader onClearAll={noop}/>
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
