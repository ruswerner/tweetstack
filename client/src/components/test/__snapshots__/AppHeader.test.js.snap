// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader should match snapshot 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#00bcd4",
      "borderRadius": "0px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "flexShrink": 0,
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "paddingLeft": 24,
      "paddingRight": 24,
      "position": "relative",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1100,
    }
  }
>
  <div
    style={
      Object {
        "marginLeft": -16,
        "marginRight": 8,
        "marginTop": 8,
        "muiPrepared": true,
      }
    }
  >
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transform": "rotate(180deg)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "white",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20h4V4h-4v16zm-6 0h4v-8H4v8zM16 9v11h4V9h-4z"
        />
      </svg>
    </button>
  </div>
  <h1
    onClick={[Function]}
    style={
      Object {
        "boxFlex": 1,
        "color": "#ffffff",
        "flex": "1",
        "fontSize": 24,
        "fontWeight": 400,
        "height": 64,
        "letterSpacing": 0,
        "lineHeight": "64px",
        "margin": 0,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingTop": 0,
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
  >
    TWEETSTACK
  </h1>
  <div
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": -16,
        "marginTop": 8,
        "muiPrepared": true,
      }
    }
  >
    <button
      disabled={false}
      id="clear-all"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "#ffffff",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "marginTop": 7,
          "minWidth": 88,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          Clear All
        </span>
      </div>
    </button>
  </div>
</div>
`;
